src/hooks/useIdleForTooLong.ts

```diff
@@ -14,13 +14,18 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with the library. If not, see <http://www.gnu.org/licenses/>.
 
-import { useEffect, useState, useCallback } from 'react'
+import { useEffect, useState, useCallback, useContext } from 'react'
 
+import { GlobalContext } from '../App'
 import { useInterval } from '../utils/hooks'
-import { walletIdleForTooLongThreshold } from '../utils/settings'
 
 const useIdleForTooLong = (onIdleForTooLong: () => void) => {
   const [lastInteractionTime, setLastInteractionTime] = useState(new Date().getTime())
+  const {
+    settings: {
+      general: { walletLockTimeInMinutes }
```

Comment:
Conceptually, this hook is agnostic regarding the action that will take place once the timeout is reached. We are passing a callback to it (that will lock the wallet) so that we don't need to pass the `wallet` variable and `setWallet` function as arguments. I don't like that we load the `walletLockTimeInMinutes` in this hook, cause now this hook is coupled with the "locking" feature. I would recommend that we pass a second argument to this hook, called `timeout`, and load the settings in the place where the hook is used.

What do you think?

Reply:
Technically speaking, what I did here is simply a variable rename. The actual correlation is pretty much the same as before, except semantically now have the term "lock" in the new variable.

I didn't say anything about the hook during the first review - I'm not against it - but as of today I don't see a case where we would use it for something else than locking the wallet. I mean, we could do other actions when locking the wallet, but I guess that the whole logic associated to the idle threshold being reached should probably be written in a single place for clarity sake.

So anyway: TLDR, the coupling doesn't bother me, having a generic hook might be a tiny bit over engineered, but now that we it, we should keep it generic. I'll make the change.

Reply:
I agree that a hook is probably overengineering, I just wanted to extract the logic out of the App component but neither a new component was an option (since I wasn't gonna render anything), nor a utility function (since I wanted to use state). Was there another alternative other than a hook?

Reply:
Making it generic is the "overengineering" part, but I mean... not a big deal at all, it took me 10 secs.

The problem with hooks is the inability to make it conditional. In our case, if the user doesn't define a lock time, ideally we shouldn't execute everything that's in the hook, but it's not always easy nor optimal to do so. See https://github.com/alephium/alephium-wallet/pull/89/commits/1e6200064c2de1b62d0d462b3752678b79b6559e

It's easier to skip executing logic with a renderless component, but less clean / more weird IMO. So yeah. Don't really know a perfect way.